### PR TITLE
Feature/node generator template

### PR DESCRIPTION
**Pull request by MiniMarker**
Ferdig med den store overføringsjobben + skrevet om docs til å handle mer om templaten og minde om den faktiske implementasjonen av prosjektet.

### CODE REVIEW

File: docusaurus/express/API/express.mdx [hunk 1]
```diff
@@ -6,47 +6,44 @@ sidebar_position: 1
...
-
-`npm install express`
-
-* [@types/express](https://www.npmjs.com/package/@types/express) - Typings for Express.js.
-
-`npm install -D @types/express`
-
 Getting started with an Express app is fairly easy and can be done by a couple of lines of code. This is great if you want to test something, but larger enterprise applications require a more robust structural layered design. 
 
 You can find the easy to follow getting started guide at Express' homepage: ["Express Hello World"](https://expressjs.com/en/starter/hello-world.html). But in larger enterprise applications a more structured system is needed.
```
Comment (JonasKs) [inline docusaurus/express/API/express.mdx:11]: Repeating text.

> something, but **larger enterprise applications require a more robust structural layered design.**

> You can find the easy to follow getting started guide at Express' homepage: "Express Hello World". **But in larger enterprise applications a more structured system is needed.**

File: docusaurus/express/API/Security/CORS.mdx [hunk 2]
```diff
@@ -8,30 +8,9 @@ sidebar_position: 3
 Cross-origin resource sharing (CORS) is a mechanism that allows restricted resources on a web page to be requested from another domain outside the domain from which the first resource was served.
```
Comment (JonasKs) [inline docusaurus/express/API/Security/CORS.mdx:8]: I'd mention that this policy is enforced by web browsers, but configured by the backend server.

File: docusaurus/express/API/Security/Authorization.mdx [hunk 3]
```diff
@@ -5,123 +5,58 @@ slug: /API/Security/Authorization
...
 
-* `tid` or tenantId: The authenticated users origin tenant ID.
-* `acct` or account: Users account status in tenant.
-* `roles`: List of roles for this application.
+:::info
+This assumes that you have an Azure AD application registration created for your project.
 
-## Roles
+If you dont have created one, head over to this guide: [Azure | Application Registrations](../Azure/ApplicationRegistrations).
+:::
```
Comment (JonasKs) [inline docusaurus/express/API/Security/Authorization.mdx:18]: I'd rewrite this to be a short little notice, something like this: 


> This section assume that you have created an [Azure AD application registration](<link to the docs you're linking to>).

**Review by JonasKs — COMMENTED** [review]
### Feedback on files not commited in this PR:
File: `express/API/Swagger`
> ~But w~Writing an OpenAPI documentation is a tedious task

File: `express/API/Swagger`  
I will only comment this once, but this is true for multiple places:
bilde
could be an info block:
bilde


File: `express/API/Swagger`  
> ~But t~This might break the default redirection logic provided by us.

File: `express/API/Azure/Event-Grid`   
Entire file is outdated and should be rewritten/deleted.


#### Other feedback 
I've created my app, but I'm not really in a position to run it. I think it should be crystal clear on how to run the application, where to browse it etc. Herman made me write [this](https://create.intility.app/fastapi/setup/poetry) section, and I wish there was something similar in this documentation, on start commands, docker run commands etc. (I'd love a terminal alternative too, as I'm not using VSCode)

There is a section about TypeScript, but it feels optional? The introduction says `Express TypeScript`. Shouldn't these steps be included by default?

And as a general feedback I think you should consider structuring the docs a bit differently. Suggestion:
- Introduction
- Creating very necessary parts, such as Azure AD stuff
- How to run docker and how to run the webserver
- TypeScript specifics
- How to implement other things, such as Metro, security etc.

**Review by h3rmanj — CHANGES_REQUESTED** [review]
Good job! Added some comments/nitpicks here and there.

The directory should be renamed from `generator-node` to `express`, to match the general repository structure and naming.

In the docs, I'd like Node.js, TypeScript and application wording of the template to be replaced with Express and API.

As Jonas mentioned the docs should have it's structure revised.
After the Introduction/Installation/Getting Started page, there should be a section with steps that are immediately needed to get the project up and running.
I recommend trying the structure .NET and FastAPI has, with Set Up, Topics and Advanced.

The TypeScript section feels a little out of place as well, we should maybe consider moving that to Intility/Documentation.

File: .github/workflows/publish-nodejs.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Publish to NPM
+
+on:
+  push:
+    branches:
+      - main
```
Comment (h3rmanj) [inline .github/workflows/publish-nodejs.yml:9]: This should use tags, with wildcard for express;

```yml
on:
  push:
    tags:
      - 'express-v*'
```

**Comment by MiniMarker** [comment]
@h3rmanj 
> The directory should be renamed from `generator-node` to `express`, to match the general repository structure and naming.

This is not possible due to yeoman's naming conventions, i can rename the generator to `generator-express` if thats better

**Comment by h3rmanj** [comment]
> @h3rmanj 
> 
> > The directory should be renamed from `generator-node` to `express`, to match the general repository structure and naming.
> 
> 
> 
> This is not possible due to yeoman's naming conventions, i can rename the generator to `generator-express` if thats better
> 
> 
> 
> 

But the package is not published from the root? And the name will still be `generator-express` as long as it's the package name.

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply package-lock.json:1]: It is needed in the ci file to build the yeoman generator

[the remark this answers is not in the document]
Comment (h3rmanj) [reply package-lock.json:1]: The root one? Are you sure?

[the remark this answers is not in the document]
Comment (MiniMarker) [reply package-lock.json:1]: Yes. Yeoman needs the package `yeoman-generator` installed and the pipeline runs `npm ci`

File: generator-node/generators/app/index.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+  install() {
+  }
+
+  end() {
+    this.log(`📦 Initializing your Git repository!`)
+    this.spawnCommandSync("git", ["init", "--initial-branch=main"]);
+    
+    this.log(`🧶 Creating QA branch...`)
+    this.spawnCommandSync("git", ["checkout", "-b", "qa"]);
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js:143]: I can change to `dev` instead but I think it's better to initialize the project in a dev branch

File: generator-node/generators/app/index.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+      },
+      {
+        type: "confirm",
+        name: "includeDemoFolders",
+        message: "🎁 Would you like to include example endpoints from the generated project?"
+      },
+      {
+        type: "input",
+        name: "gitSshAddress",
+        message: `🚀 Insert your Git SSH Address ${chalk.gray("(Optional)")}:`,
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js:52]: Makes sense

File: generator-node/package.json [hunk 8]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@intility/generator-node",
+  "version": "0.0.5",
+  "description": "Yeoman CLI to generate a new Node.JS project based on Intility developer standards",
+  "repository": "Intility/templates/generator-node",
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/package.json]: Will fix.
https://docs.npmjs.com/cli/v8/configuring-npm/package-json#repository

File: generator-node/package.json [hunk 9]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@intility/generator-node",
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/package.json]: Yo supports scoped pacakges. Will
Make sure all places match

File: generator-node/generators/app/index.js [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    this.log("")
+  }
+
+  prompting() {
+    const prompts = [
+      {
+        type: "input",
+        name: "projectTitle",
+        message: "🏷 Please enter your projects name:",
+        default: this.appname
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js:30]: Will see if this is possible ☺️

File: generator-node/README.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+```bash
+npm install -g yo                          # Install Yeoman CLI
+npm install -g @intility/generator-node    # Install Intility Node project generator
+```
+
+Then generate your new project:
+
+```bash
+yo @intility/node                          # Use Yeoman CLI to run the template generator
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/README.md:32]: Updated all install scripts in the project

File: generator-node/generators/app/templates/README.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+npm install -g yo                          # Install Yeoman CLI
+npm install -g @intility/generator-node    # Install Intility Node project generator
+```
+
+Then generate your new project:
+
+```bash
+mkdir <projectName> && cd <projectName>    # Create a new project folder
+yo @intility/node                          # Use Yeoman CLI to run the template generator
+```
```
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/templates/README.md]: Will update

File: generator-node/generators/app/index.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js:30]: I can't seem to make this work.

Can fix this in an later update?

File: generator-node/generators/app/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js:143]: Any thoughts?

File: generator-node/generators/app/index.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (h3rmanj) [reply generator-node/generators/app/index.js:30]: Of course 😃

File: generator-node/generators/app/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (h3rmanj) [reply generator-node/generators/app/index.js:143]: I'd prefer it to just initialize the git repository, and let it be up to the developer how they want to take it from there. However if you feel like it's needed and as long as it's documented well enough, `dev` is miles better than `qa`.

[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js:143]: Ait, will remove this

File: generator-node/generators/app/templates/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/templates/README.md]: fixed

File: generator-node/generators/app/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (MiniMarker) [reply generator-node/generators/app/index.js]: fixed

**Comment by MiniMarker** [comment]
I think that was all of your comments fixed

**Comment by MiniMarker** [comment]
Da merger jeg da?

**Pull request merged**